internal/pubsub/pubsub.go

```diff
@@ -42,23 +42,15 @@ type Message struct {
 // be sent again to the associated Subscription. This method blocks until
 // the message has been confirmed as acknowledged on the server, or failure
 // occurs.
-func (m *Message) Ack(ctx context.Context) error {
+func (m *Message) Ack() {
 	// Send the message back to the subscription for ack batching.
-	mec := msgErrChan{
-		msg:     m,
-		errChan: make(chan error),
-	}
 	// size is an estimate of the size of a single AckID in bytes.
 	const size = 8
-	if err := m.sub.ackBatcher.AddWait(ctx, mec, size); err != nil {
-		return err
-	}
-	return <-mec.errChan
-}
-
-type msgErrChan struct {
-	msg     *Message
-	errChan chan error
+	go func() {
+		if err := m.sub.ackBatcher.Add(m, size); err != nil {
+			panic(err)
```

Comment:
`panic` doesn't seem like a good idea in a library... What are the reasons for `Add` to return an `error`? Maybe just

`go m.sub.ackBatcher.Add(m, size)`

?

Reply:
The reasons are `ErrOversizedItem` or `ErrOverflow`.